parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetRecordReader.java

```diff
@@ -198,6 +206,10 @@ private void initializeInternalReader(ParquetInputSplit split, Configuration con
     }
 
     MessageType fileSchema = footer.getFileMetaData().getSchema();
+    Map<String, String> fileMetaData = footer.getFileMetaData().getKeyValueMetaData();
+    readContext = readSupport.init(new InitContext(configuration, InternalParquetRecordReader
+        .toSetMultiMap(fileMetaData), fileSchema));
```

Comment:
Not sure whether there is an existing utility class that is suitable to contain this logic, so instead of creating a new utility class specific to this logic how about we add a static utility method to `ReadSupport`?